src/jobflow_remote/cli/batch.py

```diff
@@ -31,11 +36,43 @@
 app.add_typer(app_batch)
 
 
+def _check_exception_dev_version(exc: ValidationError):
+    """
+    Helper function to handle the change in the "jobs" type in the BatchDoc object.
+
+    Initially defined as dict was switched to a list. Users may have this in their DB
+    if using development version.
+
+    Parameters
+    ----------
+    exc
+        The pydantic ValidationError exception to be verified.
+    """
+    try:
+        # capture all possible to avoid that if some keys are not present it
+        # fails with a confusing message.
+        val_errors = exc.errors()
+        if (
+            exc.error_count() == 1
+            and val_errors[0]["loc"] == ("jobs",)
+            and isinstance(val_errors[0]["input"], dict)
+        ):
+            exit_with_error_msg(
+                "It seems that you have used a development version of jobflow-remote. The internal format"
+                "of the batch jobs have changed before the release. Please run 'jf batch fix-batch-doc-jobs-dict'"
+                " to upgrade the database content."
```

Comment:
Maybe add something like this ?

" to upgrade the database content (note that this command is hidden from the help and can only be discovered through this message)."